src/MetalPerformanceShaders/MPSStateResourceList.cs

```diff
@@ -21,54 +21,41 @@ public partial class MPSStateResourceList {
 		{
 			if (descriptors is null)
 				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (descriptors));
-			if (descriptors.Length > 10)
-				throw new ArgumentException ("Only 10 parameters are currently supported.");
+			if (descriptors.Length > Messaging.MaxVarArgs + 1)
+				throw new ArgumentException ($"Only {Messaging.MaxVarArgs + 1} parameters are currently supported.");
 
-			var arr = new IntPtr [10];
-			for (int i = 0; i < descriptors.Length; ++i) {
-				if (descriptors [i] is null)
-					throw new ArgumentException ($"'{nameof (descriptors)}[{i}]' was null.");
-				arr [i] = descriptors [i].Handle;
-			}
+			var firstValue = descriptors [0].GetNonNullHandle ($"{nameof (descriptors)} [0]");
+			var varArgs = new IntPtr [descriptors.Length - 1];
+			for (int i = 1; i < descriptors.Length; ++i)
```

Comment:
Why exactly ++i vs i++. I see that it comes from the old code, but i'm wondering since AFAIK those too are the same when working in a for loop.